collection.js

```diff
@@ -0,0 +1,312 @@
+const { toTinsel, omit } = require('./utils.js');
+
+function hasNestedProperties(object, properties) {
+  for (const key of Object.keys(object)) {
+    if (properties.indexOf(key) !== -1) {
+      return true;
+    }
+    if (object[key] && ((typeof object[key]) === 'object')) {
+      if (hasNestedProperties(object[key], properties)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+module.exports = function (baseClass) {
+  class TinselCollection extends baseClass {
+    get __emulated() {
+      return true;
+    }
+
+    // conversion APIs
+    aggregate(op1 /* , op2... */, callback) {
+      const last = arguments.length && arguments[arguments.length - 1];
+      // Bring back support for operations as a variable number of
+      // parameters rather than as an array
+      if (Array.isArray(op1)) {
+        const options = arguments[1];
+        if (options && ((typeof options) === 'object')) {
+          if (options.cursor) {
+            // Behaves 100% like 3.x, so pass straight through
+            return super.aggregate(...Array.prototype.slice(arguments));
+          }
+        }
+        // Normal: array of aggregate stages
+        if ((typeof last) === 'function') {
+          // 2.x driver took a callback or returned a promise for results directly,
+          // 3.x driver always returns a cursor so convert back to results
+          return super.aggregate(op1).toArray(last);
+        } else {
+          // Both 2.x and 3.x return a cursor in the absence of a callback,
+          // despite documentation implying you must explicitly ask
+          // for a cursor
+          return super.aggregate(op1);
+        }
+      } else {
+        // Positional arguments as aggregate stages (2.x supports, 3.x does not)
+        if ((typeof last) === 'function') {
+          // 2.x driver supported passing a callback rather than
+          // returning a cursor, 3.x driver does not
+          return super.aggregate(...Array.prototype.slice.call(arguments, 0, arguments.length - 1)).toArray(last);
+        } else {
+          // Both 2.x and 3.x return a cursor in the absence of a callback,
+          // despite documentation implying you must explicitly ask
+          // for a cursor
+          return super.aggregate(...Array.prototype.slice.call(arguments));
+        }
+      }
+    };
+
+    // initializeUnorderedBulkOp(options) {
```

Comment:
Why is this still here if it won't be included? 🤔

Reply:
If for some reason we use this, that's the only way to override the bulk operation response with our custom classes.